Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -621,9 +621,66 @@ However, this is not a concern on iOS. As discussed in the section [Testing on t
 
 #### Overview
 
-Obfuscation is the process of transforming code and data to make it more difficult to comprehend. It is an integral part of every software protection scheme. What's important to understand is that obfuscation isn't something that can be simply turned on or off. Programs can be made incomprehensible, in whole or in part, in many ways and to different degrees.
+Obfuscation is a process of transforming the original algorithms, data structures into a form that is difficult to disassemble and understand but the program after the modification preserves the original functionality. Both code and data can be obfuscated, so that an attacker can mostly rely on dynamic analysis.
```

Comment:
```suggestion
Obfuscation is a process of transforming code into a form that is difficult to disassemble and understand and is an integral part of every software protection scheme. The application preserves the original functionality after obfuscation. What's important to understand is that obfuscation isn't something that can be simply turned on or off. Programs can be made incomprehensible, in whole or in part, in many ways and to different degrees. 
```

Let's talk about code in general and make it more generic. 

I would also suggest to keep the original last two sentences. Just because obfuscation is used doesn't mean that you need to rely mostly on dynamic analysis. It highly depends on the degree of usage of obfuscation and what your use case is to RE an app.